src/main/java/pl/cleankod/exchange/entrypoint/AccountController.java

```diff
@@ -24,36 +28,36 @@ public AccountController(FindAccountAndConvertCurrencyUseCase findAccountAndConv
         this.findAccountUseCase = findAccountUseCase;
     }
 
-    @GetMapping(path = "/{id}")
-    public ResponseEntity<Account> findAccountById(@PathVariable String id, @RequestParam(required = false) String currency) {
+    @Get("/{id}")
+    public HttpResponse<Account> findAccountById(@PathVariable String id, @Nullable @QueryValue String currency) {
+        MDC.put("requestId", UUID.randomUUID().toString());
```

Comment:
Is there maybe a more generic way of adding MDC without having to repeat it for every endpoint?